### ISSUE TITLE
How to get loss from model.val() 

### ISSUE THREAD

**Issue by Trirat2543** [ultralytics/ultralytics#6838]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How to extract val/box_loss  ,  val/cls_loss and val/dfl_loss  from model.val() mode ?

**Comment by glenn-jocher** [comment]
@Trirat2543 apologies for any confusion earlier. If you're looking to obtain validation losses without running `model.train()`, you can use the `model.fit()` method with `--task val`. This way, you'll perform a standalone validation run which will compute the losses you're interested in.

After running `model.fit()` with validation, the returned `results` object will contain the validation losses, which you can access accordingly. Please check our documentation on the Val mode for more specific guidance on accessing these metrics. 📈🛠️

**Comment by silvia1999** [comment]
@glenn-jocher Hi, I wanted to get the specific value of the PR curve via the `model.val()` command, so I ran the following command: `metrics = net.val(data='path of my yaml')`. But I'm getting the following error：
![image](https://github.com/ultralytics/ultralytics/assets/92805279/d1a40a32-2572-471d-a408-472bd0c9fadd)
I installed ipython and still can't fix it. What should I do?
Thank you for your attention.

**Comment by glenn-jocher** [comment]
@silvia1999 when you run `model.val()` in YOLOv8, it returns a `Metrics` object that contains various validation metrics, including the losses you're interested in. To extract the specific losses such as `val/box_loss`, `val/cls_loss`, and `val/dfl_loss`, you can access the attributes of the `Metrics` object.

Here's an example of how you can do this:

```python
from ultralytics import YOLO

# Load your model (ensure it's the correct path to your model)
model = YOLO('path/to/your/model.pt')

# Run validation
metrics = model.val()

# Access the losses
box_loss = metrics.box.loss
cls_loss = metrics.cls.loss
dfl_loss = metrics.dfl.loss

# Print the losses
print(f"Validation Box Loss: {box_loss}")
print(f"Validation Class Loss: {cls_loss}")
print(f"Validation Distribution Focal Loss: {dfl_loss}")
```

Please note that the exact attribute names to access the losses may vary depending on the version of the YOLOv8 codebase and the specific implementation details. The example above assumes a typical structure based on the information provided. If the attribute names are different in the version you are using, you may need to adjust the code accordingly.

If you encounter any issues or the attributes do not exist, you can check the structure of the `Metrics` object by printing its attributes or by looking at the source code where the `Metrics` class is defined.

**Comment by trianta2** [comment]
> @silvia1999 when you run `model.val()` in YOLOv8, it returns a `Metrics` object that contains various validation metrics, including the losses you're interested in. To extract the specific losses such as `val/box_loss`, `val/cls_loss`, and `val/dfl_loss`, you can access the attributes of the `Metrics` object.
> 
> Here's an example of how you can do this:
> 
> ```python
> from ultralytics import YOLO
> 
> # Load your model (ensure it's the correct path to your model)
> model = YOLO('path/to/your/model.pt')
> 
> # Run validation
> metrics = model.val()
> 
> # Access the losses
> box_loss = metrics.box.loss
> cls_loss = metrics.cls.loss
> dfl_loss = metrics.dfl.loss
> 
> # Print the losses
> print(f"Validation Box Loss: {box_loss}")
> print(f"Validation Class Loss: {cls_loss}")
> print(f"Validation Distribution Focal Loss: {dfl_loss}")
> ```
> 
> Please note that the exact attribute names to access the losses may vary depending on the version of the YOLOv8 codebase and the specific implementation details. The example above assumes a typical structure based on the information provided. If the attribute names are different in the version you are using, you may need to adjust the code accordingly.
> 
> If you encounter any issues or the attributes do not exist, you can check the structure of the `Metrics` object by printing its attributes or by looking at the source code where the `Metrics` class is defined.

Stop 👏 Making 👏 Stuff 👏 Up.
